File: scripts/addUrlIds.py
#!/usr/bin/env python3
"""Script to add urlId field to all tracks in tracks.json"""

import json
import os

# Get the path to tracks.json
script_dir = os.path.dirname(os.path.abspath(__file__))
tracks_path = os.path.join(script_dir, '..', 'data', 'tracks.json')

# Read the tracks file
try:
    with open(tracks_path, 'r', encoding='utf-8') as f:
        tracks = json.load(f)
except Exception as e:
    print(f'Error reading tracks.json: {e}')
    exit(1)

# Add urlId to each track if it doesn't exist
modified = False
for key, track in tracks.items():
    if 'urlId' not in track:
        track['urlId'] = key
        modified = True
        print(f'Added urlId "{key}" to track: {track.get("title", "Unknown")}')

if modified:
    # Write back to file with pretty formatting
    try:
        with open(tracks_path, 'w', encoding='utf-8') as f:
            json.dump(tracks, f, indent=2, ensure_ascii=False)
        print('\n✓ Successfully added urlId fields to all tracks!')
        print(f'✓ File saved: {tracks_path}')
    except Exception as e:
        print(f'Error writing tracks.json: {e}')
        exit(1)
else:
    print('All tracks already have urlId fields.')

File: scripts/addUrlIds.js
// Script to add urlId field to all tracks in tracks.json
const fs = require('fs');
const path = require('path');

const tracksPath = path.join(__dirname, '..', 'data', 'tracks.json');

// Read the tracks file
fs.readFile(tracksPath, 'utf8', (err, data) => {
  if (err) {
    console.error('Error reading tracks.json:', err);
    process.exit(1);
  }

  try {
    const tracks = JSON.parse(data);
    let modified = false;

    // Add urlId to each track if it doesn't exist
    for (const [key, track] of Object.entries(tracks)) {
      if (!track.urlId) {
        track.urlId = key;
        modified = true;
        console.log(`Added urlId "${key}" to track: ${track.title}`);
      }
    }

    if (modified) {
      // Write back to file with pretty formatting
      fs.writeFile(tracksPath, JSON.stringify(tracks, null, 2), 'utf8', (err) => {
        if (err) {
          console.error('Error writing tracks.json:', err);
          process.exit(1);
        }
        console.log('\n✓ Successfully added urlId fields to all tracks!');
        console.log(`✓ File saved: ${tracksPath}`);
      });
    } else {
      console.log('All tracks already have urlId fields.');
    }
  } catch (parseErr) {
    console.error('Error parsing tracks.json:', parseErr);
    process.exit(1);
  }
});
